src/components/MessagesList/PinnedMessage/PinnedMessage.vue

```diff
@@ -0,0 +1,195 @@
+<!--
+  - SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+  - SPDX-License-Identifier: AGPL-3.0-or-later
+-->
+<script setup lang="ts">
+
+import { t } from '@nextcloud/l10n'
+import { computed, onMounted } from 'vue'
+import { useRoute } from 'vue-router'
+import { useStore } from 'vuex'
+import NcActionButton from '@nextcloud/vue/components/NcActionButton'
+import NcListItem from '@nextcloud/vue/components/NcListItem'
+import IconClose from 'vue-material-design-icons/Close.vue'
+import IconPinOff from 'vue-material-design-icons/PinOffOutline.vue'
+import AvatarWrapper from '../../AvatarWrapper/AvatarWrapper.vue'
+import { AVATAR } from '../../..//constants.ts'
+import { useGetToken } from '../../../composables/useGetToken.ts'
+import { EventBus } from '../../../services/EventBus.ts'
+import { useActorStore } from '../../../stores/actor.ts'
+import { useSharedItemsStore } from '../../../stores/sharedItems.ts'
+import { parseToSimpleMessage } from '../../../utils/textParse.ts'
+
+const store = useStore()
+const actorStore = useActorStore()
+const route = useRoute()
+const token = useGetToken()
+const sharedItemsStore = useSharedItemsStore()
+const conversation = computed(() => store.getters.conversation(token.value))
+const richSubline = computed(() => {
+	if (!pinnedMessage.value) {
+		return ''
+	}
+	return parseToSimpleMessage(
+		pinnedMessage.value.message,
+		pinnedMessage.value.messageParameters,
+	)
+})
+
+// Array of all pinned messages
+const pinnedMessages = computed(() => {
+	if (!sharedItemsStore.sharedItems(token.value).pinned) {
+		return []
+	}
+	return Object.values(sharedItemsStore.sharedItems(token.value).pinned) || []
+})
+
+// The pinned message to be displayed (the latest one that is not hidden)
+const pinnedMessage = computed(() => {
+	if (!pinnedMessages.value.length) {
+		return null
+	}
+	const item = pinnedMessages.value.find((item) => +item.id === conversation.value.lastPinnedId)
+
+	if (!item) {
+		// Id not found (e.g: deleted, expired), return most recent pinned message
+		const fallbackId = sharedItemsStore.findMostRecentPinnedMessageId(token.value)
+		return pinnedMessages.value.find((item) => +item.id === fallbackId)
+	}
+	return item.id !== conversation.value.hiddenPinnedId ? item : null
+})
+
+// Display name for the pinned message
+const isPinnerSameAsAuthor = computed(() => pinnedMessage.value?.metaData?.pinnedActorId === pinnedMessage.value?.actorId
+	&& pinnedMessage.value?.metaData?.pinnedActorType === pinnedMessage.value?.actorType)
+
+const isPinnerSameAsCurrentUser = computed(() => pinnedMessage.value?.metaData?.pinnedActorId === actorStore.actorId
+	&& pinnedMessage.value?.metaData?.pinnedActorType === actorStore.actorType)
+
+const displayNameDetails = computed(() => {
+	if (isPinnerSameAsAuthor.value) {
+		return pinnedMessage.value?.actorDisplayName || ''
+	}
+	if (isPinnerSameAsCurrentUser.value) {
```

Comment:
Same usage and relations, should be in one computed

```suggestion
const displayNameDetails = computed(() => {
	if (pinnedMessage.value?.metaData?.pinnedActorId === pinnedMessage.value?.actorId
		&& pinnedMessage.value?.metaData?.pinnedActorType === pinnedMessage.value?.actorType) {
		return pinnedMessage.value?.actorDisplayName || ''
	}
	if (actorStore.checkIfSelfIsActor({
		actorId: pinnedMessage.value?.metaData?.pinnedActorId,
		actorType: pinnedMessage.value?.metaData?.pinnedActorType,
	})) {
```